sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -766,6 +769,17 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
     }
   }
 
+  /**
+   * Ensure that the `singleVariantColumn` option cannot be used if there is also a user specified
+   * schema.
+   */
+  private def validateSingleVariantColumn(): Unit = {
```

Comment:
I think we only need to do it once in `def load`?

Reply:
If I understand correctly, in that case we'll also have to add the checks to the dataset readers, e.g. `def json(jsonDataset: Dataset[String]): DataFrame`, since these paths don't go through `load`.

Reply:
I see, 2 places is not a big difference from 3 places.